tests/integration/defs/perf/test_perf.py

```diff
@@ -565,99 +602,126 @@ def to_cmd(self, working_dir: str) -> List[str]:
+            "s_moe_backend":
+            self.moe_backend,
+            "l_moe_max_num_tokens":
+            self.moe_max_num_tokens,
+            # cuda_graph_config
+            "b_enable_padding":
+            self.enable_padding,
+            "l_cuda_graph_max_batch_size":
+            self.cuda_graph_max_batch_size,
+            # kv_cache_config
+            "s_kv_cache_dtype":
+            self.kv_cache_dtype,
+            "b_enable_block_reuse":
+            self.enable_block_reuse,
+            "d_free_gpu_memory_fraction":
+            self.free_gpu_memory_fraction,
+            # cache_transceiver_config
+            "s_cache_transceiver_backend":
+            self.cache_transceiver_backend,
+            "l_cache_transceiver_max_tokens_in_buffer":
+            self.cache_transceiver_max_tokens_in_buffer,
+            # speculative_config
+            "s_spec_decoding_type":
+            self.spec_decoding_type,
+            "l_num_nextn_predict_layers":
+            self.num_nextn_predict_layers,
+            "s_eagle3_layers_to_capture":
+            ",".join(map(str, self.eagle3_layers_to_capture)),
+            "l_max_draft_len":
+            self.max_draft_len,
+            "s_speculative_model_dir":
+            self.speculative_model_dir,
+            "s_server_log_link":
+            "",
         }
+        return db_data
 
     def generate_extra_llm_api_config(self) -> str:
         """Generate extra-llm-api-config.yml content"""
-        config_lines = [
-            f"tensor_parallel_size: {self.tp}",
-            f"moe_expert_parallel_size: {self.ep}",
-            f"pipeline_parallel_size: {self.pp}",
-            f"max_num_tokens: {self.max_num_tokens}",
-            f"enable_attention_dp: {str(self.enable_attention_dp).lower()}",
-            f"disable_overlap_scheduler: {str(self.disable_overlap_scheduler).lower()}",
-            f"stream_interval: {self.stream_interval}",
-            f"attn_backend: {self.attention_backend}",
-            f"enable_chunked_prefill: {str(self.enable_chunked_prefill).lower()}",
-            "cuda_graph_config:",
-            f"  enable_padding: {str(self.enable_padding).lower()}",
-            f"  max_batch_size: {self.max_batch_size}",
-            "kv_cache_config:",
-            f"  dtype: {self.kv_cache_dtype}",
-            f"  free_gpu_memory_fraction: {self.free_gpu_memory_fraction}",
-            f"  enable_block_reuse: {str(self.enable_block_reuse).lower()}",
-            "print_iter_log: false",
-        ]
-
-        # Add moe_config if moe_backend is specified
-        if self.moe_backend:
-            config_lines.append("moe_config:")
-            config_lines.append(f"  backend: {self.moe_backend}")
-            if self.moe_max_num_tokens:
-                config_lines.append(
-                    f"  max_num_tokens: {self.moe_max_num_tokens}")
+        # Make a copy to avoid modifying the original
+        config_data = dict(self.extra_llm_api_config_data)
```

Comment:
```suggestion
        import copy
        config_data = copy.deepcopy(self.extra_llm_api_config_data)
```

We need to deepcopy, else the nested dict speculative_config will be modifying the original one IIUC